provider/ec2/iam.go

```diff
@@ -359,7 +359,11 @@ func listInstanceProfilesForController(
 		})
 
 		if err != nil {
-			return nil, errors.Annotatef(
+			var opHTTPErr *awshttp.ResponseError
+			if stderrors.As(err, &opHTTPErr) && opHTTPErr.HTTPStatusCode() == http.StatusForbidden {
```

Comment:
we have this here and another below, and we also have
`opHTTPErr.HTTPStatusCode() == http.StatusNotFound` in 2 places
can we introduce 2 helper methods and use those
`isNotFoundError(err error) bool`
`isUnauthorisedError(err error) bool`

Reply:
Fixed